src/praisonai-agents/praisonaiagents/toolsets.py

```
@@ -157,6 +157,54 @@ def resolve_toolset(self, name: str) -> List[str]:
+            model: Active model id used to resolve the harness profile.
+
+        Returns:
+            List of unique tool names, edit primitives ordered by preference.
+        """
+        tools = self.resolve_toolset(name)
+        if not model:
+            return tools
+        try:
+            from .model_harness import resolve_harness
+            preferred = resolve_harness(model).preferred_edit_format
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)